Remove swap transfer data

gemdev111 opened the pull request:
closes #880

gemdev111 committed:
Refactor swap flow to use callback for navigation

Replaces the use of swapTransferData for navigation with an onSwap callback in SwapSceneViewModel.
Updates navigation logic in SelectedAssetNavigationStack and SwapNavigationView to use the new callback, simplifying state management and decoupling navigation from the view model's internal state.

gemdev111 committed:
Update core

gemdev111 committed:
ci comma fix

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Updates SwapNavigation view init state

gemcoder21 commented on Features/Swap/Sources/ViewModels/SwapSceneViewModel.swift:
```diff
@@ -62,7 +62,8 @@ public final class SwapSceneViewModel {
         walletsService: WalletsService,
-        swapQuotesProvider: SwapQuotesProvidable
+        swapQuotesProvider: SwapQuotesProvidable,
+        onSwap: ((TransferData) -> Void)? = nil
```
```suggestion
        onSwap: TransferDataAction = nil
```

there is something like this

gemcoder21 commented on Features/Swap/Sources/ViewModels/SwapSceneViewModel.swift:
```diff
@@ -382,25 +384,26 @@ extension SwapSceneViewModel {
-            quote: quote,
-            quoteData: nil
+        onSwap?(
+            SwapTransferDataFactory.swap(
```
```suggestion
            SwapTransferDataFactory.swap(
```

because it's a factory, it need initializer from `SwapTransferDataFactory.from(quote:)`, quote has access to from/to asset

gemcoder21 committed:
Refactor onSwap to use TransferDataAction type

Replaces the onSwap closure type from ((TransferData) -> Void)? to TransferDataAction for improved type clarity and consistency in SwapSceneViewModel.
